### PR TITLE
LibWeb: Make `testharness` WPT tests run and fix some calc-related crashes

### PR DESCRIPTION
**Pull request by stelar7**
It still needs a small hack to manually mark the document readystate as complete before executing an async script, but other than that there is nothing stopping the tests from running

### CODE REVIEW

File: Userland/Libraries/LibWeb/CSS/StyleValues/CalculatedStyleValue.cpp [hunk 1]
```diff
@@ -38,7 +39,13 @@ static double resolve_value(CalculatedStyleValue::CalculationResult::Value value
         [](Number const& number) { return number.value(); },
         [](Angle const& angle) { return angle.to_degrees(); },
         [](Frequency const& frequency) { return frequency.to_hertz(); },
-        [&context](Length const& length) { return length.to_px(*context).to_double(); },
+        [&context](Length const& length) {
+            if (!context.has_value()) {
+                dbgln_if(LIBWEB_CSS_DEBUG, "Length value has no context, returning 0.0");
+                return 0.0;
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/CSS/StyleValues/CalculatedStyleValue.cpp:45]: This feels like it's covering up a bug elsewhere. Why are we trying to resolve a length without a resolution context?

Comment (stelar7) [reply Userland/Libraries/LibWeb/CSS/StyleValues/CalculatedStyleValue.cpp]: Theres lots of places where we need to pass the context, and currently dont. but for now ill leave this change out and we can slowly fix those places.

File: Userland/Libraries/LibWeb/Layout/LayoutState.cpp [hunk 2]
```diff
@@ -246,8 +246,12 @@ void LayoutState::UsedValues::set_node(NodeWithStyleAndBoxModelMetrics& node, Us
...
                 return true;
             }
-            resolved_definite_size = adjust_for_box_sizing(size.calculated().resolve_length(node)->to_px(node), size, width);
-            return true;
+
+            auto maybe_length = size.calculated().resolve_length(node);
+            if (maybe_length.has_value()) {
+                resolved_definite_size = adjust_for_box_sizing(maybe_length.value().to_px(node), size, width);
+                return true;
+            }
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/Layout/LayoutState.cpp:254]: This looks like a workaround. What's the problem here?
Do we have a `calc()` value that doesn't resolve to a length?
If so, how did it end up in one of the size properties (width/height/min-*)?

Also please add tests when doing bug fixes so we can understand better what they are meant to be fixing. :)

**Review by awesomekling — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibWeb/Layout/LayoutState.cpp (hunk 2 above)
Comment (stelar7) [reply Userland/Libraries/LibWeb/Layout/LayoutState.cpp]: Tried to reproduce this on latest commit, but i couldnt.. So it seems its been fixed by something else already 🤷

**Comment by stelar7** [comment]
not stale

**Pull request closed without merging**